File: highlight/src/main/java/me/rerere/highlight/languages/Languages.kt
package me.rerere.highlight.languages

import me.rerere.highlight.core.Language
import me.rerere.highlight.languages.bash.bash
import me.rerere.highlight.languages.c.c
import me.rerere.highlight.languages.cmake.cmake
import me.rerere.highlight.languages.cpp.cpp
import me.rerere.highlight.languages.csharp.csharp
import me.rerere.highlight.languages.css.css
import me.rerere.highlight.languages.dart.dart
import me.rerere.highlight.languages.diff.diff
import me.rerere.highlight.languages.dockerfile.dockerfile
import me.rerere.highlight.languages.go.go
import me.rerere.highlight.languages.glsl.glsl
import me.rerere.highlight.languages.ini.ini
import me.rerere.highlight.languages.java.java
import me.rerere.highlight.languages.javascript.javascript
import me.rerere.highlight.languages.json.json
import me.rerere.highlight.languages.kotlin.kotlin
import me.rerere.highlight.languages.latex.latex
import me.rerere.highlight.languages.lua.lua
import me.rerere.highlight.languages.markdown.markdown
import me.rerere.highlight.languages.php.php
import me.rerere.highlight.languages.powershell.powershell
import me.rerere.highlight.languages.properties.properties
import me.rerere.highlight.languages.python.python
import me.rerere.highlight.languages.rust.rust
import me.rerere.highlight.languages.ruby.ruby
import me.rerere.highlight.languages.sql.sql
import me.rerere.highlight.languages.swift.swift
import me.rerere.highlight.languages.typescript.typescript
import me.rerere.highlight.languages.xml.xml
import me.rerere.highlight.languages.yaml.yaml

/**
 * Every grammar bundled with the highlighter.
 *
 * Each entry builds a fresh mode tree: compilation mutates modes in place, mirroring `highlight.js`.
 */
internal fun builtinLanguages(): List<Language> = listOf(
    json(),
    ini(),
    cmake(),
    go(),
    glsl(),
    yaml(),
    bash(),
    dockerfile(),
    javascript(),
    typescript(),
    xml(),
    css(),
    dart(),
    java(),
    kotlin(),
    latex(),
    lua(),
    powershell(),
    properties(),
    python(),
    c(),
    cpp(),
    csharp(),
    sql(),
    diff(),
    markdown(),
    rust(),
    ruby(),
    php(),
    swift(),
)

File: highlight/tools/languages.mjs
// The grammars bundled by the Kotlin highlighter, with the aliases it answers to.
//
// The fixture generator registers exactly these with highlight.js so that sub-language lookups
// resolve the same way on both sides: a `subLanguage` naming a grammar we do not ship must fall
// back to plain text in the golden fixtures too.

export const LANGUAGES = [
  { name: 'json', module: 'json', aliases: ['json5'] },
  { name: 'ini', module: 'ini', aliases: ['ini'] },
  { name: 'cmake', module: 'cmake', aliases: [] },
  { name: 'go', module: 'go', aliases: [] },
  { name: 'glsl', module: 'glsl', aliases: [] },
  { name: 'yaml', module: 'yaml', aliases: [] },
  { name: 'bash', module: 'bash', aliases: ['shell'] },
  { name: 'dockerfile', module: 'dockerfile', aliases: [] },
  { name: 'javascript', module: 'javascript', aliases: ['js', 'jsx', 'mjs', 'cjs'] },
  { name: 'typescript', module: 'typescript', aliases: ['ts', 'tsx', 'mts', 'cts'] },
  {
    name: 'xml',
    module: 'xml',
    aliases: ['html', 'xhtml', 'rss', 'atom', 'xjb', 'xsd', 'xsl', 'plist', 'wsf', 'svg'],
  },
  { name: 'css', module: 'css', aliases: [] },
  { name: 'dart', module: 'dart', aliases: [] },
  { name: 'java', module: 'java', aliases: ['jsp'] },
  { name: 'kotlin', module: 'kotlin', aliases: ['kt', 'kts'] },
  { name: 'latex', module: 'latex', aliases: ['tex'] },
  { name: 'lua', module: 'lua', aliases: ['pluto'] },
  { name: 'ruby', module: 'ruby', aliases: ['rb', 'gemspec', 'podspec', 'thor', 'irb'] },
  { name: 'powershell', module: 'powershell', aliases: ['pwsh', 'ps', 'ps1'] },
  { name: 'properties', module: 'properties', aliases: [] },
  { name: 'python', module: 'python', aliases: ['py', 'gyp', 'ipython'] },
  { name: 'c', module: 'c', aliases: ['h'] },
  {
    name: 'cpp',
    module: 'cpp',
    aliases: ['cc', 'c++', 'h++', 'hpp', 'hh', 'hxx', 'cxx'],
  },
  { name: 'csharp', module: 'csharp', aliases: ['cs', 'c#'] },
  { name: 'sql', module: 'sql', aliases: [] },
  { name: 'diff', module: 'diff', aliases: ['patch'] },
  { name: 'markdown', module: 'markdown', aliases: ['md', 'mkdown', 'mkd'] },
  { name: 'rust', module: 'rust', aliases: ['rs'] },
  { name: 'php', module: 'php', aliases: [] },
  { name: 'swift', module: 'swift', aliases: [] },
]
